<!-- Starts AGV drive node, assuming controllers are already there -->

<launch>

    <!-- Enable/disable visualization -->
    <arg name="plots" default="False" />

    <!-- Run agv_drive_sim, and remaps outputs-->
    <node   pkg="agv_drive_sim"
            type="agv_drive_sim"
            name="agv_drive_sim"
            args=""
            output="screen">
            <remap 
                from="/agv_drive_sim/trajectory"
                to="/trajectory" />
            <remap 
                from="/agv_drive_sim/fl_steer"
                to="/agv/agv_fl_steer_controller/command"/>
            <remap 
                from="/agv_drive_sim/fr_steer"
                to="/agv/agv_fr_steer_controller/command"/>
            <remap 
                from="/agv_drive_sim/bl_steer"
                to="/agv/agv_bl_steer_controller/command"/>
            <remap 
                from="/agv_drive_sim/br_steer"
                to="/agv/agv_br_steer_controller/command"/>                
            <remap 
                from="/agv_drive_sim/fl_wheel_rate"
                to="/agv/agv_fl_drive_controller/command"/>                
            <remap 
                from="/agv_drive_sim/fr_wheel_rate"
                to="/agv/agv_fr_drive_controller/command"/>                
            <remap 
                from="/agv_drive_sim/bl_wheel_rate"
                to="/agv/agv_bl_drive_controller/command"/>                
            <remap 
                from="/agv_drive_sim/br_wheel_rate"
                to="/agv/agv_br_drive_controller/command"/>                
                
    </node>
    
    <group if="$(arg plots)">
        <node 
            name="fl_steer_plot"
            pkg="rqt_plot"
            type="rqt_plot"
            args="/agv/agv_fl_steer_controller/command /agv/joint_states/position[6]">
        </node>

        <node 
            name="fr_steer_plot"
            pkg="rqt_plot"
            type="rqt_plot"
            args="/agv/agv_fr_steer_controller/command /agv/joint_states/position[7]">
        </node>
    </group>        
    
    
</launch>
